torch/fx/graph.py

```diff
@@ -922,11 +927,25 @@ def emit_node(node : Node):
             # have been emitted. To continue to have valid Python code, emit a
             # single pass statement
             body.append('pass\n')
+        if self._in_spec is not None:
+            assert(self._orig_args is not None)
+            orig_vars = self._orig_args
+            has_orig_self = (orig_vars[0] == 'self')
```

Comment:
Under what condition would this not be true?

Reply:
If the original function we're tracing is a function and not a module.

The general logic here is that we flatten out all the input arguments. If we pass in a function, however, all of the input arguments when we trace != all of the input arguments of the function we end up with.

I might be able to move this logic to `symbolic_trace`, but that's the general problem.

Reply:
I thought about it a bit, but I don't see any real way to make this logic less complicated without (potentially) a significant refactor.